lib/core/src/sdk.rs

```diff
@@ -1006,14 +1006,20 @@ impl LiquidSdk {
 
         let swap = match self.persister.fetch_send_swap_by_invoice(invoice)? {
             Some(swap) => match swap.state {
+                Created => swap,
```

Comment:
This looks correct to me but I think it is not a complete fix. Here are some points that need to be addressed IMO:
 1. It looks like we don't check before broadcast that we already broadcasted. I think we must check it, we also need to check the current swap status before sending lockup and ensure it is in initial state (Created).
 2. I can see races when the notification job gets notify and attempt to handle state transition in parallel to the app, it seems that if both handle the invoice.set state at the same time, double broadcast may happen. Perhaps we should disable broadcasting the lockup from the job or think on another solution to handle this point.

Reply:
> It looks like we don't check before broadcast that we already broadcasted.

We check if it's present locally, but perhaps we should also make an onchain query

> ensure it is in initial state (Created)

Agreed

> More specifically, the requirement for broadcasting a lockup should be only the process where the send was invoked.

Agreed